src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordCopyCtor.cs

```diff
@@ -45,11 +48,26 @@ internal override void GenerateMethodBodyStatements(SyntheticBoundNodeFactory F,
             var param = F.Parameter(Parameters[0]);
             foreach (var field in ContainingType.GetFieldsToEmit())
             {
-                if (!field.IsStatic)
+                if (!field.IsStatic && field.ContainingType.Equals(ContainingType))
```

Comment:
> && field.ContainingType.Equals(ContainingType) [](start = 35, length = 47)

Why is this check added, ```GetFieldsToEmit``` is not supposed to return fields that do not belong to the ```ContainingType```.